flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/JsonDebeziumSchemaSerializer.java

```diff
@@ -156,13 +161,25 @@ public DorisRecord serialize(String record) throws IOException {
             return null;
         }
 
-        if (firstLoad) {
-            schemaChange.init(recordRoot);
-            firstLoad = false;
+        Map<String, String> tableMapping = schemaChange.getTableMapping();
+        if (initSchemaChange(firstLoad, tableMapping)) {
+            schemaChange.init(recordRoot, initTableSet);
+            this.firstLoad = false;
         }
+
         return dataChange.serialize(record, recordRoot, op);
     }
 
+    private boolean initSchemaChange(boolean firstLoad, Map<String, String> tableMapping) {
+        if (firstLoad) {
+            return true;
+        }
```

Comment:
Do we need to determine the `firstLoad` of each table?

Reply:
sounds like a good idea